Pronto-MIA/Pronto-MIA/DataAccess/Managers/FirebaseMessagingManager.cs

```diff
@@ -178,5 +232,42 @@ private IFirebaseMessagingAdapter GetInstance()
             return new FirebaseMessagingAdapter(
                 FirebaseMessaging.DefaultInstance);
         }
+
+        /// <summary>
+        /// Method to remove all fcmTokens which are invalid according
+        /// to the api responses from the database.
+        /// </summary>
+        /// <param name="tokens">The tokens which have been used in the api
+        /// request. It is important, that the tokens are still in the same
+        /// order as they have been sent to the api.</param>
+        /// <param name="responses">The Responses sent by the api. It is
+        /// important, that the responses are still in the same order as they
+        /// were received from the api.</param>
+        /// <returns>An empty task which does not have to be awaited since this
+        /// is only a cleanup operation.</returns>
+        private HashSet<string> GetInvalidTokens(
```

Comment:
Wouldn't this belong to `FirebaseTokenManager` ?

Reply:
It is a function that gets the invalid tokens from a message response. That is why it is located within this service.